### ISSUE TITLE
Discord integration something went wrong

### ISSUE THREAD

**Issue by Johannp02180** [healthchecks/healthchecks#576]
Hello, when settings up discord, I can access the link login, and select my server and channel. It then redirects back to healthcheck and I get a something went wrong error:
<img width="1158" alt="Screen Shot 2021-10-19 at 2 19 46 PM" src="https://user-images.githubusercontent.com/11222092/137968273-b51c9d3c-9784-434a-937a-9a6323b2e50b.png">

I'm not sure if this in the logs will help:
`[pid: 301|app: 0|req: 22/50] 10.253.0.5 () {36 vars in 971 bytes} [Tue Oct 19 18:19:35 2021] GET /integrations/add_discord/?code=niRz7OE46ONlmtDO4ztVJkvsNf5XLv&state=b6Ha5yiDz2FrAoyUbotyQog2rgyKqDciKEbHPbTebFE&guild_id=661424797208674314 => generated 0 bytes in 118 msecs (HTTP/1.1 302) 10 headers in 610 bytes (1 switches on core 0)`

**Comment by Enissay** [comment]
@Johannp02180 I've got the same, how have you fixed it ?

**Comment by cuu508** [comment]
It produces the "Something went wrong." message during the OAuth2 flow, when exchanging auth token for an access token, if the response from the server does not contain an "access_token" field.

@Enissay are you seeing this error on a self-hosted instance? If so, to better understand the issue, add a debug statement in hc/front/views.py, `add_discord_complete` function:

```python
    result = curl.post("https://discordapp.com/api/oauth2/token", data)

    doc = result.json()
    print(result.status_code, doc)  # <--- ADD THIS
    if "access_token" in doc:
        channel = Channel(kind="discord", project=project)
        channel.value = result.text
        channel.save()
        channel.assign_all_checks()
        messages.success(request, "The Discord integration has been added!")
    else:
        messages.warning(request, "Something went wrong.")

    return redirect("hc-channels", project.code)
```

**Comment by Enissay** [comment]
> are you seeing this error on a self-hosted instance? If so, to better understand the issue, add a debug statement
Yes, a docker instance. Sadly it seems to require a restart which looses the change since it is not persistent :-/
Or, how can I restart the app it inside the running container ?

**Comment by Enissay** [comment]
Finally it was solved.
I had to reset `DISCORD_CLIENT_SECRET` and get a new one. Which means the old one somehow was wrong >_>
I thought to do it yesterday but then I told myself it was unlikely...
Anyhow, I hope this would help future lost souls...
Thank you @cuu508 for you fast reactivity and putting me on the right path <3
